src/main/java/Storage.java

```diff
@@ -0,0 +1,87 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+class Storage {
+
+    public static ArrayList<Task> tasks = new ArrayList<Task>();
+    public static File file = new File("data/duke.txt");
+    public void createFile() throws IOException {
+        File dir = new File("data/");
+        if (!dir.exists()) {
+            dir.mkdir();
+        }
+
+        File savedTasks = file;
+        if (!savedTasks.exists()) {
+            savedTasks.createNewFile();
+        }
+    }
+    public void writeTasks( ) throws IOException {
+        File destination = file;
+        StringBuilder textToAdd = new StringBuilder();
+
+        FileWriter fw = new FileWriter(destination);
+        for (Task curr : tasks) {
+            if (curr instanceof Todo) {
+                textToAdd.append("T ")
+                        .append(curr.isDone() ? "1 " : "0 ")
+                        .append(curr.getName())
+                        .append(System.lineSeparator());
+            } else if (curr instanceof Deadline) {
+                textToAdd.append("D ")
+                        .append(curr.isDone() ? "1 " : "0 ")
+                        .append(curr.getName())
+                        .append(System.lineSeparator())
+                        .append(((Deadline) curr).getBy())
+                        .append(System.lineSeparator());
+            } else if (curr instanceof Event) {
+                textToAdd.append("E ")
+                        .append(curr.isDone() ? "1 " : "0 ")
+                        .append(curr.getName())
+                        .append(System.lineSeparator())
+                        .append(((Event) curr).getAt())
+                        .append(System.lineSeparator());
+            }
+        }
+        fw.write(textToAdd.toString());
+        fw.close();
+    }
+
+
+
+    public ArrayList<Task> readTasks( ) throws FileNotFoundException {
+
+        File source = file;
+        Scanner sc = new Scanner(source);
+        while (sc.hasNext()) {
+            String current = sc.nextLine();
+            String[] splits = current.split(" ", 3);
+            switch (splits[0]) {
```

Comment:
Please remember to follow the module's switch statements standard. 😉

Reply:
You can manually change the indentation in intelliJ's settings too!